components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2450,6 +2492,196 @@ public void updateGroup(Group oldGroup, Group newGroup) throws CharonException {
+                    } else if (StringUtils.equals(SCIMConstants.GroupSchemaConstants.MEMBERS,
+                            patchOperation.getAttributeName())) {
+                        memberOperations.add(patchOperation);
+                    }
+                }
+            }
+
+            Collections.reverse(displayNameOperations);
+
+            if (CollectionUtils.isNotEmpty(displayNameOperations)) {
+                newGroupName = (String) displayNameOperations.get(0).getValues();
+                setGroupDisplayName(currentGroupName, newGroupName);
+            }
+
+            Collections.sort(memberOperations);
+            Set<String> addedMembers = new HashSet<>();
+            Set<String> deletedMembers = new HashSet<>();
+            Set<Object> newlyAddedMemberIds = new HashSet<>();
+
+            for (PatchOperation memberOperation : memberOperations) {
+                if (memberOperation.getValues() instanceof Map) {
+                    Map<String, String> memberObject = (Map<String, String>) memberOperation.getValues();
+                    prepareAddedRemovedMemberLists(addedMembers, deletedMembers, newlyAddedMemberIds, memberOperation,
+                            memberObject);
+                } else if (memberOperation.getValues() instanceof List) {
+                    List<Map<String, String>> memberOperationValues =
+                            (List<Map<String, String>>) memberOperation.getValues();
+                    for (Map<String, String> memberObject : memberOperationValues) {
+                        prepareAddedRemovedMemberLists(addedMembers, deletedMembers, newlyAddedMemberIds,
+                                memberOperation, memberObject);
+                    }
+                }
+            }
+
+            String userStoreDomainForGroup = IdentityUtil.extractDomainFromName(newGroupName);
+            Set<String> temporaryMembers = new HashSet<>();
+
+            if (isNotInternalOrApplicationGroup(userStoreDomainForGroup) && (!addedMembers.isEmpty()
+                    || !deletedMembers.isEmpty())) {
+                for (String member : addedMembers) {
+                    String username = UserCoreUtil.addDomainToName(UserCoreUtil.removeDomainFromName(member),
+                            userStoreDomainForGroup);
+                    temporaryMembers.add(username);
+                }
+
+                addedMembers.clear();
+                addedMembers.addAll(temporaryMembers);
+                temporaryMembers.clear();
+
+                for (String member : deletedMembers) {
+                    String username = UserCoreUtil.addDomainToName(UserCoreUtil.removeDomainFromName(member),
+                            userStoreDomainForGroup);
+                    temporaryMembers.add(username);
+                }
+
+                deletedMembers.clear();
+                deletedMembers.addAll(temporaryMembers);
+            }
+
+            // Check for deleted members
```

Comment:
Shall we add a `.` at the end of the comment?

Reply:
Resolved in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/280/commits/36add81b74fe1bfc479adf2b8217e00af089b8f7